backend/capellacollab/sessions/operators/k8s.py

```diff
@@ -668,28 +776,91 @@ def _create_service(
-                    ),
-                    client.V1ServicePort(
-                        name="metrics",
+                        name=name,
                         protocol="TCP",
-                        port=9118,
-                        target_port=9118,
-                    ),
+                        port=port,
+                        target_port=port,
+                    )
+                    for name, port in ports.items()
                 ],
                 selector={"app": deployment_name},
                 type="ClusterIP",
             ),
         )
         return self.v1_core.create_namespaced_service(namespace, service)
 
+    def _create_ingress(self, id, path: str, port_number: int):
+        ingress = client.V1Ingress(
+            api_version="networking.k8s.io/v1",
+            kind="Ingress",
+            metadata=client.V1ObjectMeta(
+                name=id,
+                annotations={"ingress.kubernetes.io/ssl-redirect": "false"},
+            ),
+            spec=client.V1IngressSpec(
+                rules=[
+                    client.V1IngressRule(
+                        host=host,
+                        http=client.V1HTTPIngressRuleValue(
+                            paths=[
+                                client.V1HTTPIngressPath(
+                                    path=path,
+                                    path_type="Prefix",
+                                    backend=client.V1IngressBackend(
+                                        service=client.V1IngressServiceBackend(
+                                            name=id,
+                                            port=client.V1ServiceBackendPort(
+                                                number=port_number
+                                            ),
+                                        )
+                                    ),
+                                )
+                            ]
+                        ),
+                    )
+                ]
+            ),
+        )
+        return client.NetworkingV1Api().create_namespaced_ingress(
+            namespace, ingress
+        )
+
+    def _create_openshift_route(self, id, path: str, port_number: int):
+        route_dict = {
+            "kind": "Route",
+            "apiVersion": "route.openshift.io/v1",
+            "metadata": {
+                "name": id,
+            },
+            "spec": {
+                "host": host,
+                "path": path,
+                "to": {
+                    "kind": "Service",
+                    "name": id,
+                },
+                "port": {
+                    "targetPort": port_number,
+                },
+                "tls": {
+                    "termination": "edge",
+                    "insecureEdgeTerminationPolicy": "Redirect",
+                },
+            },
+        }
+        dyn_client = DynamicClient(self.client)
```

Comment:
Please also move this client to the `__init__`. If you only really need the `v1_routes` you could also add `self.v1_routes` instead of the dynamic client

Reply:
Let's not move it, since it breaks non-openshift clusters.